src/main/java/com/nanuvem/metagui/server/rules/RulesContainer.java

```diff
@@ -0,0 +1,54 @@
+package com.nanuvem.metagui.server.rules;
+
+import java.util.List;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.nanuvem.metagui.server.api.Context;
+import com.nanuvem.metagui.server.api.Rule;
+import com.nanuvem.metagui.server.api.Widget;
+import com.nanuvem.metagui.server.context.ContextContainer;
+import com.nanuvem.metagui.server.widgets.WidgetContainer;
+
+@Component
+public class RulesContainer {
+	
+	@Autowired
+	private RuleRepository ruleRepository;
+	@Autowired
+	private ContextContainer contextContainer;
+	@Autowired
+	private WidgetContainer widgetContainer;
+	
+	private Long version = 0l;
+	
+	public List<Rule> getAllRules() {
+		return ruleRepository.findAll();
+	}
+	
+	public List<Rule> getAllRulesByVersionGreaterThan(Long version) {
+		return ruleRepository.findByVersionGreaterThan(version);
+	}
+	
+	public Rule saveRule(@Valid Rule rule) {
+		rule.setVersion(++version);
```

Comment:
You need to update version, every time this component is instantiated.
Maybe this solution is not thread safe.